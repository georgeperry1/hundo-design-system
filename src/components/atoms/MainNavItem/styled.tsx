import styled from 'styled-components';

import { MainNavItemProps } from '.';

export const ButtonElement = styled.button<MainNavItemProps>`
  width: ${({ active }) => (active ? '100%' : '80%')};
  background-color: ${({ backgroundColor }) => (backgroundColor ? backgroundColor : '#FFFF50')};
  border: none;
  color: ${({ shadowColor }) => (shadowColor ? shadowColor : '#B7B73D')};
  box-sizing: border-box;
  display: flex;
  align-items: center;
  font-size: ${({ theme }) => theme.fontSizes[4]};
  font-weight: 900;
  font-family: ${({ theme }) => theme.fonts.primary};
  justify-content: center;
  line-height: 1.75rem;
  padding: 0.75rem 1.65rem;
  position: relative;
  text-align: center;
  text-decoration: none #000000 solid;
  text-decoration-thickness: auto;
  max-width: 460px;
  position: relative;
  cursor: pointer;
  transform: rotate(1deg);
  user-select: none;
  -webkit-user-select: none;
  touch-action: manipulation;
  box-shadow: 1px 1px 0px 0px, 2px 2px 0px 0px, 3px 3px 0px 0px, 4px 4px 0px 0px, 5px 5px 0px 0px;

  &:focus {
    outline: 0;
  }
  &:active {
    box-shadow: 0px 0px 0px 0px;
    top: 5px;
    left: 5px;
  }
`;

export const ButtonWrapper = styled.div`
  width: 190px;
  padding-left: 30px;
`;

export const TextElement = styled.span`
  color: #000000;
`;

export const ImgElement = styled.img`
  width: 20px;
  height: 20px;
  margin-right: 12px;
`;
