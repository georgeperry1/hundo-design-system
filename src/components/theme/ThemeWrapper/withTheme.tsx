import React from 'react';
import { ThemeProvider } from 'styled-components';
import { lightTheme } from '../themes';

export function withTheme<P>(WrappedComponent: React.ComponentType<P>): React.FC<P> {
  const ComponentWithTheme = (props: P) => {
    return (
      <ThemeProvider theme={lightTheme}>
        <WrappedComponent {...props} />
      </ThemeProvider>
    );
  };
  return ComponentWithTheme;
}
export default withTheme;
