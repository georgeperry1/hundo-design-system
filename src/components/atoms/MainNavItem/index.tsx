import React from 'react';

import { ButtonElement, ButtonWrapper, TextElement, ImgElement } from './styled';

export type MainNavItemProps = {
  active?: boolean;
  icon?: string;
  text?: string;
  backgroundColor?: string;
  shadowColor?: string;
};

const MainNavItem: React.FC<MainNavItemProps> = (props) => (
  <ButtonWrapper>
    <ButtonElement {...props}>
      <ImgElement src={props.icon} />
      <TextElement>{props.text}</TextElement>
    </ButtonElement>
  </ButtonWrapper>
);

MainNavItem.defaultProps = {
  active: false,
};

export default MainNavItem;
