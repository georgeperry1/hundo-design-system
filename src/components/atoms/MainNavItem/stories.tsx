import { Story, Meta } from '@storybook/react';

import MainNavItem, { MainNavItemProps } from '.';

export default {
  title: 'Components/Atoms/MainNavItem',
  component: MainNavItem,
} as Meta;

export const Template: Story<MainNavItemProps> = (args) => <MainNavItem {...args}>{args.text}</MainNavItem>;

Template.args = {
  active: false,
  text: 'Profile',
  backgroundColor: '#FFFF50',
  shadowColor: '#B7B73D',
};

export const Default = Template.bind({});
