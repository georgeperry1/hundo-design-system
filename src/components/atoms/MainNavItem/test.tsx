import { render } from '@testing-library/react';

import MainNavItem from '.';
import ThemeWrapper from '../../../utils/test-theme-wrapper';

const NavItemProps = {
  text: 'Profile',
  icon: 'https://www.svgrepo.com/show/794/male-user.svg',
  backgroundColor: '#FFFF50',
  shadowColor: '#B7B73D',
};

describe('MainNav', () => {
  it('should render', () => {
    const { baseElement } = render(<ThemeWrapper TestedComponent={MainNavItem} componentProps={NavItemProps} />);

    expect(baseElement).toMatchSnapshot();
  });

  it('should show active state', () => {
    const { baseElement } = render(
      <ThemeWrapper TestedComponent={MainNavItem} componentProps={{ ...NavItemProps, active: true }} />,
    );

    expect(baseElement).toMatchSnapshot();
  });
});
