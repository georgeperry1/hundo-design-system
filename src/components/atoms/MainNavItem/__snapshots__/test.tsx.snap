// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNav should render 1`] = `
.c1 {
  width: 80%;
  background-color: #FFFF50;
  border: none;
  color: #B7B73D;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1rem;
  font-weight: 700;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 1.75rem;
  padding: 0.75rem 1.65rem;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none #000000 solid;
  text-decoration: none #000000 solid;
  -webkit-text-decoration-thickness: auto;
  text-decoration-thickness: auto;
  max-width: 460px;
  position: relative;
  cursor: pointer;
  -webkit-transform: rotate(1deg);
  -ms-transform: rotate(1deg);
  transform: rotate(1deg);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-user-select: none;
  touch-action: manipulation;
  box-shadow: 1px 1px 0px 0px,2px 2px 0px 0px,3px 3px 0px 0px,4px 4px 0px 0px,5px 5px 0px 0px;
}

.c1:focus {
  outline: 0;
}

.c1:active {
  box-shadow: 0px 0px 0px 0px;
  top: 5px;
  left: 5px;
}

.c0 {
  width: 190px;
  padding-left: 30px;
}

.c3 {
  color: #000000;
}

.c2 {
  width: 20px;
  height: 20px;
  margin-right: 12px;
}

<body>
  <div>
    <div
      class="c0"
    >
      <button
        class="c1"
      >
        <img
          class="c2"
          src="https://www.svgrepo.com/show/794/male-user.svg"
        />
        <span
          class="c3"
        >
          Profile
        </span>
      </button>
    </div>
  </div>
</body>
`;

exports[`MainNav should show active state 1`] = `
.c1 {
  width: 100%;
  background-color: #FFFF50;
  border: none;
  color: #B7B73D;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1rem;
  font-weight: 700;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 1.75rem;
  padding: 0.75rem 1.65rem;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none #000000 solid;
  text-decoration: none #000000 solid;
  -webkit-text-decoration-thickness: auto;
  text-decoration-thickness: auto;
  max-width: 460px;
  position: relative;
  cursor: pointer;
  -webkit-transform: rotate(1deg);
  -ms-transform: rotate(1deg);
  transform: rotate(1deg);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-user-select: none;
  touch-action: manipulation;
  box-shadow: 1px 1px 0px 0px,2px 2px 0px 0px,3px 3px 0px 0px,4px 4px 0px 0px,5px 5px 0px 0px;
}

.c1:focus {
  outline: 0;
}

.c1:active {
  box-shadow: 0px 0px 0px 0px;
  top: 5px;
  left: 5px;
}

.c0 {
  width: 190px;
  padding-left: 30px;
}

.c3 {
  color: #000000;
}

.c2 {
  width: 20px;
  height: 20px;
  margin-right: 12px;
}

<body>
  <div>
    <div
      class="c0"
    >
      <button
        class="c1"
      >
        <img
          class="c2"
          src="https://www.svgrepo.com/show/794/male-user.svg"
        />
        <span
          class="c3"
        >
          Profile
        </span>
      </button>
    </div>
  </div>
</body>
`;
